source/clear-linux/tutorials/spark.rst

```diff
@@ -0,0 +1,136 @@
+framework, enter:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add big-data-basic
+
+Configure Apache Spark
+**********************
+
+#. Create the configuration directory with the command:
+
+   .. code-block:: bash
+
+      sudo mkdir /etc/spark
+
+#. Copy the default templates from :file:`/usr/share/defaults/spark` to
+   :file:`/etc/spark` with the command:
+
+   .. code-block:: bash
+
+      sudo cp /usr/share/defaults/spark/* /etc/spark
+
+   .. note:: Since |CL| is a stateless system, you should never modify the
+      files under the :file:`/usr/share/defaults` directory. The software
+      updater overwrites those files.
+
+
+#. Copy the template files below to create custom configuration files:
+
+   .. code-block:: bash
+
+      sudo cp /etc/spark/spark-defaults.conf.template /etc/spark/spark-defaults.conf
+      sudo cp /etc/spark/spark-env.sh.template /etc/spark/spark-env.sh
+      sudo cp /etc/spark/log4j.properties.template /etc/spark/log4j.properties
+
+#. Edit the :file:`/etc/spark/spark-env.sh` file and add the
+   :option:`SPARK_MASTER_HOST` variable. Replace the example address below
+   with your localhost IP address. (View your IP address using the
+   :command:`hostname -I` command.)
+
+   .. code-block:: xml
+
+      SPARK_MASTER_HOST="10.300.200.100"
+
+   .. note:: This optional step enables the master's web user interface to
+      view information needed later in this tutorial.
+
+#. Edit the :file:`/etc/spark/spark-defaults.conf` file and update the
+   :option:`spark.master` variable with the :option:`SPARK_MASTER_HOST` address
+   from the previous step and port :option:`7077`.
+
+   .. code-block:: xml
+
+      spark.master    spark://10.300.200.100:7077
+
+Start the master server and a worker daemon
+*******************************************
+
+#. Start the master server using:
+
+   .. code-block:: bash
+
+      sudo /usr/share/apache-spark/sbin/./start-master.sh
+
+#. Start one worker daemon and connect it to the master using the
+   :option:`spark.master` variable defined earlier:
+
+   .. code-block:: bash
+
+      sudo /usr/share/apache-spark/sbin/./start-slave.sh spark://10.300.200.100:7077
+
+#. Open an internet browser and view the worker daemon information using
+   the master's IP address and port :option:`8080`:
+
+   .. code-block:: xml
+
+      http://10.300.200.100:8080
+
+Run the Spark wordcount example
+*******************************
+
+#. Run the wordcount example using a file on your local host and output the
+   results to a new file with the following command:
+
+   .. code-block:: bash
+
+      sudo spark-submit /usr/share/apache-spark/examples/src/main/python/wordcount.py ~/Documents/example_file > ~/Documents/results
+
+#. Open an internet browser and view the application information using
+   the master's IP address and port :option:`8080`:
+
+   .. code-block:: xml
+
+      http://10.300.200.100:8080
+
+#. View the results of the wordcount application in the results file specified
```

Comment:
Do not reference a previous step. State explicitly the name and path of the file.